core/src/main/java/com/linecorp/armeria/internal/common/stream/SurroundingPublisher.java

```diff
@@ -108,7 +108,14 @@ public void subscribe(Subscriber<? super T> subscriber, EventExecutor executor,
 
         if (!subscribedUpdater.compareAndSet(this, 0, 1)) {
             subscriber.onSubscribe(NoopSubscription.get());
-            subscriber.onError(new IllegalStateException("Only single subscriber is allowed!"));
+            if (completionFuture.isCompletedExceptionally()) {
+                completionFuture.exceptionally(cause -> {
+                    subscriber.onError(cause);
+                    return null;
+                });
+            } else {
+                subscriber.onError(new IllegalStateException("Only single subscriber is allowed!"));
+            }
```

Comment:
- If aborted (completionFuture.isCompletedExceptionally(): true)
  - then onError(abortedCause) 👈 without this, TCK test failed~!
- If already subscribed and not aborted
  - then onError(IllegalStateException)

```
// required_subscribeOnAbortedStreamMustFail test
assertThat(capturedCause.get()).isInstanceOf(AbortedStreamException.class); // should onError(AbortedException)
```

@ikhoon To pass above TCK test, I add this logic~!